### PR TITLE
assignment5

### PR DESCRIPTION
**Pull request by jy330545**

### CODE REVIEW

File: assignment5.txt [hunk 1]
```diff
@@ -0,0 +1,23 @@
+Justin Yu
+
+1. I would call it by:
+
+	int main() {
```
Comment (oldclesleycode) [inline assignment5.txt:5]: right, but you're missing one thing - which is the function declaration at the top!

Comment (oldclesleycode) [inline assignment5.txt:5]: It will return a warning w/o it.

File: assignment5.txt [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	int a, c;
+	float b;
+	int fxn(int a, float b, int c);
+	return 0; 
+	}
+
+my code assigns value to integers a and c and float b, then it calls for the parameter int fxn(int a, float b, int c).
+
+1. The difference between recursion and iteration is that: although they are both about repetition, because recursion performs a procedure inversely where as iteration performs it directly, it requires the data to be stored until it reaches the final step of the procedure and work its way back. Therefore, recursion uses a lot of storage and is typically slower, which makes it worse in most cases comparing to iteration.
```
Comment (oldclesleycode) [inline assignment5.txt:14]: Nice!

File: assignment5.txt [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+2. The function of a compiler can be separated into seven phases.
```
Comment (lesley2958) [inline assignment5.txt]: Perfect explanation!

File: assignment5.txt [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+   1. The source program is converted into sequence of tokens.
+   2. The compiler recovers the potential errors formed during the convertion.
+   3. It also analyzes the meaning of the product program.
```
Comment (oldclesleycode) [inline assignment5.txt:19]: Great explanation! The only thing is this analysis occurs through a parse tree!

File: loopFibonacci.c [hunk 5]
```diff
@@ -0,0 +1,28 @@
+// Justin Yu   I'm getting infinite loop but I have no idea why //
+
+#include <stdio.h>
+
+int main() {
+
+int first, second, next, x, i;
+first = 0;
```
Comment (oldclesleycode) [inline loopFibonacci.c:8]: Good job initializing these two values!

File: loopFibonacci.c [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+second = 1;
+
+printf("Enter the number of terms\n");
+scanf("%d", &x);
+
+printf("The terms are %d\n", x);
```
Comment (oldclesleycode) [inline loopFibonacci.c:14]: So this just prints back the number of terms wanted

File: loopFibonacci.c [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+for( i = 0; i < x; i++) {
+if (i <= 1) 
+	i = next;
+else {
```
Comment (oldclesleycode) [inline loopFibonacci.c:19]: not sure why you chose to do it this way - explain and i can give you some insight as to what went wrong

File: recurseFibonacci.c [hunk 8]
```diff
@@ -0,0 +1,28 @@
+// Justin Yu  This is a program that lists the terms of the Fibonacci sequence. //
+
+#include <stdio.h>
```
Comment (oldclesleycode) [inline recurseFibonacci.c:3]: Perfect! Nice job!

**Pull request closed without merging**